app/controllers/treatments/treatments_controller.rb

```diff
@@ -0,0 +1,40 @@
+class Treatments::TreatmentsController < ApplicationController
+
+  def create
```

Comment:
Hmmm.  Okay so it's taken me a little bit to wrap my mind around how you've used this.  Your application of template classing here will only really be immediately clear to you in writing it.

Rename this class ```BaseTreatmentController``` to be clear that it's a base class and you don't intend on using it concretely.

Rename this method ```create_treatment_and_redirect(trim_type:)``` and then use it in each ```create``` method in your sub controllers.  This will make a nice top-down design.